yt_dlp/extractor/ivxplayer.py

```diff
@@ -0,0 +1,71 @@
+from .common import InfoExtractor
+from ..utils import int_or_none, parse_iso8601, traverse_obj
+
+
+class IVXPlayerIE(InfoExtractor):
+    _VALID_URL = r'ivxplayer:(?P<video_id>\d+):(?P<player_key>\w+)'
+    _TESTS = [{
+        'url': 'ivxplayer:2366065:4a89dfe6bc8f002596b1dfbd600730b1',
+        'info_dict': {
+            'id': '2366065',
+            'ext': 'mp4',
+            'duration': 112,
+            'upload_date': '20221204',
+            'title': 'Film Indonesia di Disney Content Showcase Asia Pacific 2022',
+            'timestamp': 1670151746,
+            'thumbnail': 'https://ivx-image.ivideosmart.com/serve/image/video/2366065?width=300'
+        }
+    }]
+    _WEBPAGE_TESTS = [{
+        'url': 'https://www.cantika.com/video/31737/film-indonesia-di-disney-content-showcase-asia-pacific-2022',
+        'info_dict': {
+            'id': '2374200',
+            'ext': 'mp4',
+            'duration': 110,
+            'title': 'Serial Indonesia di Disney Content Showcase Asia Pacific 2022',
+            'timestamp': 1670639416,
+            'upload_date': '20221210',
+            'thumbnail': 'https://ivx-image.ivideosmart.com/serve/image/video/2374200?width=300'
+        }
+    }, {
+        'url': 'https://www.gooto.com/video/11437/wuling-suv-ramai-dikunjungi-di-giias-2018',
+        'info_dict': {
+            'id': '892109',
+            'ext': 'mp4',
+            'title': 'Wuling SUV Ramai Dikunjungi di GIIAS 2018',
+            'upload_date': '20180811',
+            'description': 'md5:6d901483d0aacc664aecb4489719aafa',
+            'duration': 75,
+            'timestamp': 1534011263,
+            'thumbnail': 'https://ivx-image.ivideosmart.com/serve/image/video/892109?width=300'
+        }
+    }]
+
+    def _extract_from_webpage(self, url, webpage):
+        # more info at https://player.ivideosmart.com/ivsplayer/v4/dist/js/loader.js
+        player_key, video_id = self._search_regex(
+            r'<ivs-player\s*[^>]+data-ivs-key\s*=\s*"(?P<player_key>[\w]+)\s*[^>]+\bdata-ivs-vid="(?P<video_id>[\w-]+)',
+            webpage, 'player_key, video_id', group=('player_key', 'video_id'), default=(None, ''))
+        if not player_key:
+            return
+        yield self.url_result(f'ivxplayer:{video_id}:{player_key}', ie=IVXPlayerIE)
+        raise self.StopExtraction()
```

Comment:
```suggestion
    def _extract_embed_urls(cls, url, webpage):
        # more info at https://player.ivideosmart.com/ivsplayer/v4/dist/js/loader.js
        mobj = self._search_regex(
            r'<ivs-player\s*[^>]+data-ivs-key\s*=\s*"(?P<player_key>[\w]+)\s*[^>]+\bdata-ivs-vid="(?P<video_id>[\w-]+)',
            webpage)
        if mobj:
            yield f'ivxplayer:{mobj.group("player_key")}:{mobj.group("video_id")}'
            raise self.StopExtraction()
```

Reply:
As `_extract_embed_urls become `classmethod`, didn't it will not work as `self` is undefined. As far i know, `_search_regex` is not a static method so changing `self._search_regex` to `cls._search_regex` will not work.

Reply:
my bad. Edited above